vllm/entrypoints/openai/serving_chat.py

```diff
@@ -539,11 +539,14 @@ async def chat_completion_stream_generator(
                         #   matched by partial json parsing
                         # only happens if we are NOT using guided decoding
                         if tool_parser:
-                            index = len(
-                                tool_parser.prev_tool_call_arr) - 1 if len(
-                                    tool_parser.prev_tool_call_arr) > 0 else 0
+                            tools_called = len(
+                                tool_parser.prev_tool_call_arr) > 0
+                            index = len(tool_parser.prev_tool_call_arr
+                                        ) - 1 if tools_called else 0
+                            tools_called = index > 0
```

Comment:
IMO, it is a bit confusing to set `tools_called` twice to different values.
```suggestion
                            index = len(
                                tool_parser.prev_tool_call_arr) - 1 if len(
                                    tool_parser.prev_tool_call_arr) > 0 else 0
                            tools_called = index > 0
```

Reply:
Oh thanks, that's actually a editing mistake resulting from working in a dev pod.